synthesizer/process/src/finalize.rs

```diff
@@ -104,7 +104,11 @@ impl<N: Network> Process<N> {
         lap!(timer, "Verify the number of transitions");
 
         // Construct the call graph.
-        let call_graph = self.construct_call_graph(execution)?;
+        let call_graph = if state.block_height() >= N::CONSENSUS_V3_HEIGHT {
+            Default::default()
+        } else {
+            self.construct_call_graph(execution)?
+        };
```

Comment:
This comment is a bit tangential to what's being fixed, but my fear is this approach of a `CONSENSUS_VX_HEIGHT` is quickly creating a situation where we could have double digit consensus versions and migratory logic like this in many places within the the SnarkVM codebase. This could easily lead to nontrivial errors.

At some point soon, the community should come up with some kind of migration model and spec which lays out a standard format for migration. 

I haven't thought this out in enough depth yet, but I imagine we might have some kind of migration `decorator`-style pattern (perhaps a macro? Enums that apply migration functions, etc.?) that clearly denotes a change in consensus rules, specifies the logic at each and does a standard set of checks.  

After the higher priority fixes `mainnet` the community  should do some active design to keep future updates and  fixes maintainable.

Reply:
It's a good call out. 
Your macro suggestion reminded me of the `malice` approach, maybe we can borrow the design there.
We could also enforce stricter standards around migration logic, requiring that a README is maintained in this repo, which defines the invariants around each new `CONSENSUS_V*_HEIGHT`.

Reply:
+1 very much something worth thinking about and documenting.

One design principle I'm confident about (which also went wrong in the first design of the Fee lowering PR) is that wherever consensus changes are introduced, they should be independent full self-contained functionality, and not complex derivations of each other's functionality.

Reply:
This is a wonderful point. I also have a great dislike for `CONSENSUS_VX_HEIGHT` as it will quickly grow out of control. It's definitely worth exploring cleaner solutions. There should be plenty of examples of other chains that have managed quick bug fixes, hard forks, etc. that we can learn from.

Reply:
Provable protocol team will hereby promise to do a small lit review in january.